x-pack/filebeat/input/gcs/job.go

```diff
@@ -133,11 +157,21 @@ func (j *job) processAndPublishData(ctx context.Context, id string) error {
 	defer func() {
 		err = reader.Close()
 		if err != nil {
+			j.metrics.errorsTotal.Inc()
 			j.log.Errorw("failed to close reader for object", "objectName", j.object.Name, "error", err)
 		}
 	}()
 
-	return j.decode(ctx, reader, id)
+	// update the source lag time metric
+	j.metrics.sourceLagTime.Update(time.Since(j.object.Updated).Milliseconds())
```

Comment:
Use nanoseconds for any durations. This is consistent with durations defined in ECS.